Improvements for M365 Dashboard Lisp

1zun4secondary opened the pull request:
Features to be added:
- [x] Add speed modes (double tap on button)
- [x] Add lock mode with beeping and braking (double press while braking)
- [x] Add min-speed feature (makes it more secure)
- [x] Add shutdown feature (turn it off by long press and back on by single tap)
- [x] Add battery in idle feature
- [x] Add separate ADC version

Fixes to be done (Probably my own issues because no one else has it):
- [x] ~~Figure out why 0x64 packets are not being read. (on my setup)~~ (Can be ignored due to the fact that we do not have to receive any 0x64 packets to sent our own 0x64 back)
- [x] ~~Figure out why button reading is randomly~~ (can be fixed with 470R resistor between 3.3v and RX)

I have read the CLA Document and I hereby sign the CLA
**I have read the CLA Document and I hereby sign the CLA**

1zun4 committed:
improved m365 dash

added min-speed feature
added show-batt-in-idle feature
improved throttle and brake values

1zun4 committed:
not needed

1zun4 committed:
improved m365 dash

added speed modes on double press (still has to implement rpm limit)
added off and on when pressing long
added lock (double press when braking)

1zun4 committed:
fixed off and lock mode

1zun4 commented:
recheck

1zun4 commented:
**I have read the CLA Document and I hereby sign the CLA**

1zun4 commented:
recheck

1zun4 commented:
recheck

1zun4 commented:
I have read the CLA Document and I hereby sign the CLA

1zun4 commented:
recheck

1zun4 commented:
> **CLA Assistant Lite bot:** Thank you for your submission, we really appreciate it. Like many open-source projects, we ask that you sign our [Contributor License Agreement](https://github.com/vedderb/vesc_tool/blob/master/cla_vesc_tool.md) before we can accept your contribution. You can sign the CLA by just posting a Pull Request Comment same as the below format.
> 
> _**I have read the CLA Document and I hereby sign the CLA**_
> 
> You can retrigger this bot by commenting **recheck** in this Pull Request

recheck

1zun4 committed:
cleanup

1zun4 committed:
renamed locked to lock

1zun4 committed:
few comments

Axel070719 commented:
I have read the CLA Document and I hereby sign the CLA

Axel070719 commented:
Recheck

1zun4 committed:
improved m365 dash

improved button responsiveness
fixed not turning on by single tap
added beep feedback
pushing while locked will now make beep
(testing ADC override - will add it very soon)

1zun4 committed:
improved m365 dash

added adc version
fully support speed modes now
cleanup

1zun4 committed:
added missing progn

1zun4 commented:
I have read the CLA Document and I hereby sign the CLA

1zun4 commented:
recheck

1zun4 commented:
https://user-images.githubusercontent.com/69533520/173804915-3ef6adf2-6026-400d-8658-1dc1feb2f66b.mp4

https://user-images.githubusercontent.com/69533520/173804955-1bd4c5dd-21b4-4a9e-8de8-e61fb45013ac.mp4


https://user-images.githubusercontent.com/69533520/173804991-12a35381-9bcb-4bb0-93b6-2d898aa54ebf.mp4

1zun4 committed:
fixed accidentally dividing watts

1zun4 commented:
Hello, so far the script has been working great, but after updating my bldc firmware from https://github.com/m365fw/bldc/commit/1a723ad8e2c4f1ecc7f82b75f196dcac7813cef7 to https://github.com/vedderb/bldc/commit/805006f03e486506145e6faa1f334d47fe9875d8

1zun4 committed:
Improve throttle and brake handling

Added throttle deadzone
Added brake deadzone
Optimized throttle and brake values
Added throttle-in and brake-in variables for calibration
Added brake min-speed and fixed noise when pressing brake on 0km/h
Fixed throttle and brake values going over 1 (not really a issue but it is better)

1zun4 committed:
fixed lock function

1zun4 committed:
Revert "fixed lock function"

This reverts commit 3b74a0671da3396911b2d80c18870ceafdac03b9.
